src/stores/media/single-result.ts

```diff
@@ -49,32 +59,48 @@ export const useSingleResultStore = defineStore('single-result', {
       return mediaItem
     },
 
-    async fetchMediaItem(type: SupportedMediaType, id: string) {
+    /**
+     * If the result is available in the media store (from search results or related media),
+     * we can re-use it. Otherwise, we fetch it from the API.
+     * The media objects from the search results currently don't have filetype or filesize properties.
```

Comment:
Why reuse it? 🤔 I'm not sure if we gain much from this optimization as the search endpoint doesn't return all the fields we need in the single result page.

Reply:
I didn't know that the search endpoint doesn't return all the fields when I implemented this. I think there are only `filetype` and `filesize` fields that are not returned currently. I think re-using still improves the performance of single result loading.